### DISCUSSION TITLE
Z-axis doesn't show tick values in volume plot?

### DISCUSSION THREAD

**Discussion by jgollub** [pyvista/pyvista#4075] [Q&A]
I'm making a volume plot and wanting to show the labels for the 3-axis. X and Y axes are labeled, but Z is not. How do I get the Z ticks to be labeled with numbers? I'm using the following code:

import pyvistaqt as pvqt
pl = pvqt.BackgroundPlotter()
pl.add_volume(mesh, scalars = 'values', cmap = 'copper', opacity = np.array([0, *np.linspace(.05,1, 200)]))
pl.background_color = 'white'

pl.view_xy(negative = True) #camera

labels =  dict(xlabel = 'X', ylabel = 'Y', zlabel = 'Z')

pl.show_bounds(
    grid= 'front',
    location ='outer',
    all_edges =True,
    **labels,
    color = 'k',
    render = True,
    )
pl.add_axes(**labels, color = 'k', x_color = 'k', y_color = 'k', z_color = 'k')
pl.show()

<img width="882" alt="Screenshot 2023-02-28 at 2 45 36 PM" src="https://user-images.githubusercontent.com/5776556/221998990-ea345588-ad1f-46c5-bb9d-9d6bf4e18468.png">

**Comment by banesullivan** [comment]
We've had a lot of issues with the bounds axes... I wonder if the changes in this pull request improve this for you?: https://github.com/pyvista/pyvista/pull/3977

**Comment by jgollub** [comment]
yeah, that fixes the issue. Thanks! When will it get integrated into the
main branch? Thanks.

On Wed, Mar 1, 2023 at 7:58 AM Bane Sullivan ***@***.***>
wrote:

> We've had a lot of issues with the bounds axes... I wonder if the changes
> in this pull request improve this for you?: #3977
> <https://github.com/pyvista/pyvista/pull/3977>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/pyvista/pyvista/discussions/4075#discussioncomment-5168294>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABMCJLEFHBJCIYQXALKOHXTWZ5W2PANCNFSM6AAAAAAVLKDJFQ>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Jonah Gollub, PhD
Research Scientist | Metacept | CMIP | Duke University | C: 858-231-9367
